GVRf/Framework/framework/src/main/java/org/gearvrf/GVRSwitch.java

```diff
@@ -0,0 +1,146 @@
+/* Copyright 2015 Samsung Electronics Co., LTD
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gearvrf;
+
+import java.lang.reflect.Method;
+import java.util.List;
+
+
+/**
+ * A switch node is a component that will display only
+ * one child of a scene object.
+ * 
+ * It can be used to select from among several versions of an object.
+ * When the GVRSwitch component is attached to a GVRSceneObject
+ * it sets all but one of the scene object's children to be invisible.
+ * You can control which child is visible by setting the switch index.
+ * 
+ * @see GVRBehavior
+ */
+public class GVRSwitch extends GVRBehavior
+{
+    static private long TYPE_SWITCH = (System.currentTimeMillis() & 0xfffffff);
+    protected int mSwitchIndex = -1;
+    protected GVRSceneObject mSelected = null;
```

Comment:
I removed it. Originally I was going to track the currently selected one but then I realized it would not be very resilient against changes to the switch children array.